nil/services/synccommittee/proofprovider/task_handler.go

```diff
@@ -13,38 +15,60 @@ import (
 
 type HandlerTaskStorage interface {
 	AddTaskEntries(ctx context.Context, tasks ...*types.TaskEntry) error
+	GetActiveTaskCount(ctx context.Context) (int, error)
 }
 
 type taskHandler struct {
-	taskStorage HandlerTaskStorage
-	resultSaver TaskResultSaver
-	skipRate    int
-	taskNum     int
-	clock       clockwork.Clock
-	logger      logging.Logger
+	taskStorage    HandlerTaskStorage
+	resultSaver    TaskResultSaver
+	skipRate       int
+	taskNum        int
+	maxActiveTasks int
+	clock          clockwork.Clock
+	logger         logging.Logger
 }
 
 func newTaskHandler(
 	taskStorage HandlerTaskStorage,
 	resultSaver TaskResultSaver,
 	skipRate int,
+	maxActiveTasks int,
 	clock clockwork.Clock,
 	logger logging.Logger,
 ) api.TaskHandler {
 	return &taskHandler{
-		taskStorage: taskStorage,
-		resultSaver: resultSaver,
-		skipRate:    skipRate,
-		taskNum:     0,
-		clock:       clock,
-		logger:      logger,
+		taskStorage:    taskStorage,
+		resultSaver:    resultSaver,
+		skipRate:       skipRate,
+		maxActiveTasks: maxActiveTasks,
+		taskNum:        0,
+		clock:          clock,
+		logger:         logger,
 	}
 }
 
+func (h *taskHandler) IsReadyToHandle(ctx context.Context) (bool, error) {
+	active, err := h.taskStorage.GetActiveTaskCount(ctx)
+	if err != nil {
+		return false, err
+	}
+	return active <= h.maxActiveTasks, nil
+}
+
 func (h *taskHandler) Handle(ctx context.Context, executorId types.TaskExecutorId, task *types.Task) error {
 	if task.TaskType != types.ProofBatch {
 		return types.NewTaskErrNotSupportedType(task.TaskType)
 	}
+
+	// Check if max limit of tasks is not exceeded
+	ready, err := h.IsReadyToHandle(ctx)
```

Comment:
When limit archived service should not fetch new tasks, otherwise the tasks will be skipped.
Need set this condition here: https://github.com/NilFoundation/nil/blob/46acd00066b9994b13175d48602368f329b2ebcf/nil/services/synccommittee/internal/executor/task_executor.go#L89